### PR TITLE
Refactor how where clauses are handled. 

### PR DESCRIPTION
**Pull request by daamsie**
This allows date filtering when getting total amounts.

### CODE REVIEW

File: includes/data/class-charitable-campaign-donations-db.php [hunk 1]
```diff
@@ -206,12 +206,15 @@ public function get_total( $statuses = array() ) {
 
 			list( $status_clause, $parameters ) = $this->get_donation_status_clause( $statuses );
 
+			if ( ! empty( $status_clause ) ) {
+				$where_sql = 'WHERE' . $status_clause;
```
Comment (ericnicolaas) [inline includes/data/class-charitable-campaign-donations-db.php:210]: Looks to me like there should be a space after WHERE here.

File: includes/data/class-charitable-campaign-donations-db.php [hunk 2]
```diff
@@ -732,12 +822,14 @@ public function count_donations_by_status( $statuses ) {
...
+			}
 
 			$sql = "SELECT COUNT( * )
-                    FROM {$wpdb->prefix}charitable_campaign_donations cd
-                    INNER JOIN $wpdb->posts p ON p.ID = cd.donation_id
-                    WHERE 1 = 1
-                    $status_clause;";
+					FROM {$wpdb->prefix}charitable_campaign_donations cd
+					INNER JOIN $wpdb->posts p ON p . ID = cd . donation_id
+					$sql_where;";
```
Comment (ericnicolaas) [inline includes/data/class-charitable-campaign-donations-db.php:832]: Just on changes like this and the one in the `get_total()` method above, is the main purpose here just to be consistent with using the `$sql_where` variable? Or is there something functionally different here? 

Also, is `p . ID` right? Shouldn't that be `p.ID`?

**Review by ericnicolaas — CHANGES_REQUESTED** [review]
@daamsie Just left a couple comments. Mostly it looks fine though I'll need to do some thorough testing to make sure this doesn't break anything unexpected.

File: includes/data/class-charitable-campaign-donations-db.php (hunk 2 above)
Comment (daamsie) [reply includes/data/class-charitable-campaign-donations-db.php:832]: p . ID *(and cd . donation_id) is definitely wrong  . Must have missed this particular function in testing! I had that in a couple of other places due to VS code auto formatting on save.. 

But yes, wrapping them into that $sql_where was just a case of keeping it consistent. Removing the need for the 1=1 in some queries. Makes it much easier to reuse it.

**Pull request merged**